src/mitol/payment_gateway/api.py

```diff
@@ -492,16 +495,19 @@ def get_refund_request(transaction_dict: Dict):
             Object: Refund An object of Refund class that can be used to operate the Refunds
 
         """
-        # Not doing a .get() on dict because this should break if the required keys are not present in the dictionary
-        transaction_id = transaction_dict["transaction_id"]
-        order_amount = transaction_dict["req_amount"]
-        order_currency = transaction_dict["req_currency"]
-
-        return Refund(
-            transaction_id=transaction_id,
-            refund_amount=order_amount,
-            refund_currency=order_currency,
-        )
+        try:
+            # If a required value exists we can't decide if it's valid so let the API throw that error itself
+            transaction_id = transaction_dict["transaction_id"]
+            order_amount = transaction_dict["req_amount"]
+            order_currency = transaction_dict["req_currency"]
+
+            return Refund(
+                transaction_id=transaction_id,
+                refund_amount=order_amount,
+                refund_currency=order_currency,
+            )
+        except KeyError as error:
+            raise InvalidTransactionException(error)
```

Comment:
This is passing `error` as `message`, I think you want `InvalidTransactionException() from error`